docs/tutorials/python/table.md

```diff
@@ -1,2 +1,254 @@
+```
+
+Note that we're propagating the etag from the query results. Without it, we'd get an error saying something about an "Invalid etag":
+
+```python
+table = syn.store(Table(schema, df, etag=results.etag))
+```
+
+The etag is used by the server to prevent concurrent users from making conflicting changes, a technique called optimistic concurrency. In case of a conflict, your update may be rejected. You then have to do another query and try your update again.
+
+## Changing Table Structure
+
+Adding columns can be done using the methods `Schema.addColumn` or `addColumns` on the `Schema` object:
+
+```python
+schema = syn.get("syn000000")
+bday_column = syn.store(Column(name='birthday', columnType='DATE'))
+schema.addColumn(bday_column)
+schema = syn.store(schema)
+```
+
+Renaming or otherwise modifying a column involves removing the column and adding a new column:
+
+```python
+cols = syn.getTableColumns(schema)
+for col in cols:
+    if col.name == "birthday":
+        schema.removeColumn(col)
+bday_column2 = syn.store(Column(name='birthday2', columnType='DATE'))
+schema.addColumn(bday_column2)
+schema = syn.store(schema)
+```
+
+## Table attached files
+
+Synapse tables support a special column type called 'File' which contain a file handle, an identifier of a file stored in Synapse. Here's an example of how to upload files into Synapse, associate them with a table and read them back later:
+
+```python
+# your synapse project
+import tempfile
+project = syn.get(...)
+
+# Create temporary files to store
+temp = tempfile.NamedTemporaryFile()
+with open(temp.name, "w+") as temp_d:
+    temp_d.write("this is a test")
+
+temp2 = tempfile.NamedTemporaryFile()
+with open(temp2.name, "w+") as temp_d:
+    temp_d.write("this is a test 2")
+
+# store the table's schema
+cols = [
+    Column(name='artist', columnType='STRING', maximumSize=50),
+    Column(name='album', columnType='STRING', maximumSize=50),
+    Column(name='year', columnType='INTEGER'),
+    Column(name='catalog', columnType='STRING', maximumSize=50),
+    Column(name='cover', columnType='FILEHANDLEID')]
+schema = syn.store(Schema(name='Jazz Albums', columns=cols, parent=project))
+
+# the actual data
+data = [["John Coltrane",  "Blue Train",   1957, "BLP 1577", temp.name],
+        ["Sonny Rollins",  "Vol. 2",       1957, "BLP 1558", temp.name],
+        ["Sonny Rollins",  "Newk's Time",  1958, "BLP 4001", temp2.name],
+        ["Kenny Burrel",   "Kenny Burrel", 1956, "BLP 1543", temp2.name]]
+
+# upload album covers
+for row in data:
+    file_handle = syn.uploadFileHandle(row[4], parent=project)
+    row[4] = file_handle['id']
+
+# store the table data
+row_reference_set = syn.store(RowSet(schema=schema, rows=[Row(r) for r in data]))
+
+# Later, we'll want to query the table and download our album covers
+results = syn.tableQuery(f"select artist, album, cover from {schema.id} where artist = 'Sonny Rollins'")
+test_files = syn.downloadTableColumns(results, ['cover'])
```

Comment:
I'm unsure just how much we want to push people to use this function any more, but rather have people use EntityViews in combination with files being uploaded.